docs/src/man/joins_new.md

```diff
@@ -0,0 +1,879 @@
+     │ Int64  Int64?   String?  Int64?   String?  String
+─────┼───────────────────────────────────────────────────
+   1 │     4       53  male           2  free     bad
+   2 │     5       35  male           1  free     good
+   3 │     6       53  male           2  own      good
+   4 │     7       35  male           3  rent     good
+   5 │     8       61  male           1  own      good
+   6 │     9       28  male           3  own      bad
+   7 │    10  missing  missing  missing  missing  bad
+   8 │    11  missing  missing  missing  missing  bad
+   9 │    12  missing  missing  missing  missing  good
+  10 │    13  missing  missing  missing  missing  bad
+  11 │    14  missing  missing  missing  missing  good
+```
+
+## Outer Join
+
+The output contains rows for values of the key that exist in the first (left) or second (right) 
+argument to join.
+
+If we want to have all `:id`s from both data frames, we use an `outerjoin`:
+
+```jldoctest dataframe
+julia> outerjoin(german, risk1, on = :id)
+15×6 DataFrame
+ Row │ id     Age      Sex      Job      Housing  Risk
+     │ Int64  Int64?   String?  Int64?   String?  String?
+─────┼────────────────────────────────────────────────────
+   1 │     4       53  male           2  free     bad
+   2 │     5       35  male           1  free     good
+   3 │     6       53  male           2  own      good
+   4 │     7       35  male           3  rent     good
+   5 │     8       61  male           1  own      good
+   6 │     9       28  male           3  own      bad
+   7 │     0       67  male           2  own      missing
+   8 │     1       22  female         2  own      missing
+   9 │     2       49  male           1  own      missing
+  10 │     3       45  male           2  free     missing
+  11 │    10  missing  missing  missing  missing  bad
+  12 │    11  missing  missing  missing  missing  bad
+  13 │    12  missing  missing  missing  missing  good
+  14 │    13  missing  missing  missing  missing  bad
+  15 │    14  missing  missing  missing  missing  good
+```
+
+It loads more `missing` values, but we can see all the `:id`s now. It check out the new `missing` 
+names for `:id`s 10-12. It is known as `outerjoin`.
+
+The above four joins make up the basics of data frame merging. We can remember like this:
+- `innerjoin`: The output contains rows for values of the key that exist in both the first (left) and 
+  second (right) arguments to `join`.
+- `leftjoin`: The output contains rows for values of the key that exist in the first (left) argument to 
+  `join`, whether or not that value exists in the second (right) argument.
+- `rightjoin`: The output contains rows for values of the key that exist in the second (right) argument 
+  to `join`, whether or not that value exists in the first (left) argument.
+- `outerjoin`: The output contains rows for values of the key that exist in the first (left) or second 
+  (right) argument to `join`.
+
```

Comment:
also mention at the end what happens in case of duplicates in `on` column and how non-standard values in `on column are handled (most importantly `missing`)